huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go

```diff
@@ -0,0 +1,521 @@
+		deleteAccountProduct = "dcs"
+	)
+	deleteAccountClient, err := cfg.NewServiceClient(deleteAccountProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DCS Client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	accountName := d.Get("account_name").(string)
+	deleteAccountPath := deleteAccountClient.Endpoint + deleteAccountHttpUrl
+	deleteAccountPath = strings.ReplaceAll(deleteAccountPath, "{project_id}", deleteAccountClient.ProjectID)
+	deleteAccountPath = strings.ReplaceAll(deleteAccountPath, "{instance_id}", instanceId)
+	deleteAccountPath = strings.ReplaceAll(deleteAccountPath, "{account_id}", d.Id())
+
+	deleteAccountOpt := golangsdk.RequestOpts{KeepResponseBody: true}
+
+	retryFunc := func() (interface{}, bool, error) {
+		_, err = deleteAccountClient.Request("DELETE", deleteAccountPath, &deleteAccountOpt)
+		retry, err := handleOperationError(err)
+		return nil, retry, err
+	}
+	_, err = common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
+		Ctx:          ctx,
+		RetryFunc:    retryFunc,
+		WaitFunc:     refreshDcsInstanceState(deleteAccountClient, instanceId),
+		WaitTarget:   []string{"RUNNING"},
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		DelayTimeout: 10 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+
+	if err != nil {
+		return diag.Errorf("error deleting the account (%s): %v", accountName, err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"EXECUTING"},
+		Target:       []string{"SUCCESS"},
+		Refresh:      accountTaskStatusRefreshFunc(instanceId, "DeleteAcl", deleteAccountClient),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        2 * time.Second,
+		PollInterval: 2 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for deleting the account (%s) to complete: %s", accountName, err)
+	}
+
+	return resourceDcsAccountRead(ctx, d, meta)
+}
+
+func accountStatusRefreshFunc(instanceId, accountName string, client *golangsdk.ServiceClient) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		var (
+			getAccountHttpUrl = "v2/{project_id}/instances/{instance_id}/accounts"
+		)
+		getAccountPath := client.Endpoint + getAccountHttpUrl
+		getAccountPath = strings.ReplaceAll(getAccountPath, "{project_id}", client.ProjectID)
+		getAccountPath = strings.ReplaceAll(getAccountPath, "{instance_id}", instanceId)
+
+		getAccountOpt := golangsdk.RequestOpts{KeepResponseBody: true}
+
+		getAccountResp, err := client.Request("GET", getAccountPath, &getAccountOpt)
+		if err != nil {
+			return nil, "QUERY ERROR", err
+		}
+		getAccountBody, err := utils.FlattenResponse(getAccountResp)
+		if err != nil {
+			return nil, "PARSE ERROR", err
+		}
+
+		account := utils.PathSearch(fmt.Sprintf("accounts|[?instance_id =='%s' && account_name == '%s']|[0]", instanceId, accountName),
+			getAccountBody, nil)
```

Comment:
the request path has contains instance_id, is it needed here?

Reply:
deleted instance_id